### ISSUE TITLE
Akka http 10.0.10 not applying custom timeout for client requests

### ISSUE THREAD

**Issue by pianista215** [akka/akka-http#1527]
Hi, 

Is possible that the timeout configuration is not being used in Akka 10.0.10?

```
  val akkaVersion = "2.5.6"
  val akkaHttpVersion = "10.0.10"
  //Akka
  "com.typesafe.akka" %% "akka-actor" % akkaVersion,
  "com.typesafe.akka" %% "akka-testkit" % akkaVersion % Test,

  //AkkaHttp
  "com.typesafe.akka" %% "akka-http" % akkaHttpVersion,
  "com.typesafe.akka" %% "akka-http-spray-json" % akkaHttpVersion,

  //To be compatible between Akka and Akka http
  "com.typesafe.akka" %% "akka-stream" % akkaVersion,
```

Doing:

```
val timeoutSettings =
        ConnectionPoolSettings(context.system.settings.config).withIdleTimeout(10 minutes)

      http.singleRequest(
        HttpRequest(
          method = HttpMethods.POST,
          uri = endpoint,
          entity = rqEntity
        ), settings = timeoutSettings)
```

I'm getting:
`akka.stream.scaladsl.TcpIdleTimeoutException: TCP idle-timeout encountered on connection to [localhost:9080], no bytes passed in the last 1 minute.`

**Comment by jrudolph** [comment]
There are two kinds of idle timeouts. One is for shutting down the whole pool. The other one applies to every connection. You set the pool-wide idle-timeout. The connection-level one can be set with `ConnectionPoolSettings(...).withConnectionSettings(originalSettings.withIdleTimeout(...))`.

We should at least add scaladoc to both of the `withIdleTimeout` methods distinguishing their use. A longer term change could be to rename those settings to something more unique and more descriptive like "idle-pool-timeout" and "idle-connection-timeout".

**Comment by pianista215** [comment]
Ok, I will try your solution:

`ConnectionPoolSettings(...).withConnectionSettings(originalSettings.withIdleTimeout(...))`

But may be It's a good idea for the future, decouple the timeout in other parameter in the HttpRequest case class, because now is a bit confusing for the developers that doesn't know the internals of Akka http

Thank you!

**Comment by asarkar** [comment]
@jrudolph I was about to open a ticket for this after beating my head for hours. Glad that I did the due diligence of searching for open issues first. Would you kindly confirm the following, since I've already spent hours on this exact issue?

1. The [docs](https://doc.akka.io/docs/akka-http/current/scala/http/common/timeouts.html) show 4 different settings for settings for client idle timeout. Which one applies to super pool?
2. Is it true that a value of `infinite` disables the connection pool idle timeout?
3. Is there a config for connection level idle timeout or must it be done in code as you suggested above?

**Comment by jrudolph** [comment]
> The docs show 3 different settings for client idle timeout. Which one applies to super pool?

Yep, the documentation needs to be improved not to confuse those.

`akka.http.host-connection-pool.idle-timeout` is the pool timeout.

`akka.http.client.idle-timeout` is the connection-level timeout which can be overridden with `akka.http.host-connection-pool.client.idle-timeout` for pooled connections.

Setting `infinite` should work to disable the timeout on both levels.

**Comment by asarkar** [comment]
Thank you

**Comment by aaabramov** [comment]
Here is the direct link to docs: https://doc.akka.io/docs/akka-http/current/common/timeouts.html
